src/shared/install-printf.c

```diff
@@ -57,8 +58,8 @@ static int specifier_instance(char specifier, void *data, void *userdata, char *
         if (r < 0)
                 return r;
 
-        if (!instance) {
-                instance = strdup("");
+        if (isempty(instance) && i->default_instance) {
+                instance = strdup(i->default_instance);
                 if (!instance)
                         return -ENOMEM;
         }
```

Comment:
Apparently I can't read. But for my defence at least I documented my wrong thinking in the commit message ;)